pkg/controller/nodeipam/ipam/range_allocator.go

```diff
@@ -191,10 +191,21 @@ func (r *rangeAllocator) worker(stopChan <-chan struct{}) {
 				klog.Warning("Channel nodeCIDRUpdateChannel was unexpectedly closed")
 				return
 			}
-			if err := r.updateCIDRsAllocation(workItem); err != nil {
-				// Requeue the failed node for update again.
-				r.nodeCIDRUpdateChannel <- workItem
+			if err := r.updateCIDRsAllocation(workItem); err == nil {
+				klog.V(3).Infof("Updated CIDR for %q", workItem)
+			} else {
+				klog.Errorf("Error updating CIDR for %q: %v", workItem, err)
+				if canRetry, timeout := r.retryParams(workItem.nodeName); canRetry {
+					klog.V(2).Infof("Retrying update for %q after %v", workItem, timeout)
+					time.AfterFunc(timeout, func() {
+						// Requeue the failed node for update again.
+						r.nodeCIDRUpdateChannel <- workItem
+					})
+					continue
+				}
+				klog.Errorf("Exceeded retry count for %q, dropping from queue", workItem)
 			}
+			r.removeNodeFromProcessing(workItem.nodeName)
```

Comment:
In order to do this remove, you have to have a way to re-queue on a change.  The update handling between the two controllers appears to be different. See 

https://github.com/kubernetes/kubernetes/blob/4ae11dac2e2cc8594c528a679db5861c0926b514/pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go#L117-L123

Reply:
Are you making two different points here? The requeuing issue and the difference in update handling issue?

The update handling does seem to be different and no reason not to sync so it sets the NetworkUnavailable taint.

In terms of requeueing, I think you are right in that there is a `defer r.removeNodeFromProcessing(data.nodeName)` at the top of `updateCIDRsAllocation()` that needs to go.